packages/medusa/src/api/routes/store/customers/claim-orders.ts

```diff
@@ -0,0 +1,134 @@
+import { SamlTokenUserAccountAuthMutationVariables } from "@linear/sdk/dist/_generated_documents"
+import { IsNotEmpty, IsString } from "class-validator"
+import { MedusaError } from "medusa-core-utils"
+import { EntityManager } from "typeorm"
+import { Order } from "../../../../models"
+import {
+  CustomerService,
+  EventBusService,
+  OrderService,
+} from "../../../../services"
+import TokenService from "../../../../services/token"
+
+/**
+ * @oas [post] /customers/claim-orders
```

Comment:
thought: We should be able to perform a similar action as a store administrator, such that in case the customer inquires about their previous orders, we would be able to easily:
- Navigate to the customer
- List orders by customer id
- Send a request to claim orders triggering the email flow

Reply:
I am fine to keep it out of this PR, but I think it would be a super valuable addition.